### PR TITLE
Add seasons endpoints

### PR DESCRIPTION
**Pull request by oleg1std**
Add endpoints to manage seasons

### CODE REVIEW

File: HappyTravel.Hiroshima.DirectManager/Services/AccommodationManagementService.cs [hunk 1]
```diff
@@ -72,7 +82,21 @@ public Task<Result> Remove(int accommodationId)
             return _contractManagerContext.GetContractManager()
                 .Bind(async contractManager =>
                 {
-                    await _accommodationManagementRepository.DeleteAccommodationAndRooms(contractManager.Id, accommodationId);
+                    var accommodation = await _accommodationManagementRepository.GetAccommodation(contractManager.Id, accommodationId);
```
Comment (mingazhev) [inline HappyTravel.Hiroshima.DirectManager/Services/AccommodationManagementService.cs:85]: Please consider moving such logics to local function to make code more descriptive

File: HappyTravel.Hiroshima.DirectManager/Services/AccommodationManagementService.cs [hunk 2]
```diff
@@ -161,14 +187,22 @@ private Accommodation CreateAccommodation(int contractManagerId, Models.Requests
 
         private Models.Responses.Accommodation CreateResponse(Accommodation accommodation, List<int> roomIds = null)
```
Comment (mingazhev) [inline HappyTravel.Hiroshima.DirectManager/Services/AccommodationManagementService.cs:188]: Could this function be static?

File: HappyTravel.Hiroshima.DirectManager/Services/SeasonManagementService.cs [hunk 3]
```diff
@@ -1,6 +1,142 @@
...
+            _contractManagerContext = contractManagerContext;
+            _contractManagementRepository = contractManagementRepository;
+        }
+
+
+        public Task<Result<List<Models.Responses.Season>>> Get(int contractId)
+        {
+            return _contractManagerContext.GetContractManager()
+                .Ensure(contractManager => DoesContractBelongToContractManager(contractId, contractManager.Id),
+                    $"Failed to get the contract by {nameof(contractId)} '{contractId}'")
```
Comment (mingazhev) [inline HappyTravel.Hiroshima.DirectManager/Services/SeasonManagementService.cs:30]: Please add an error text that describes error more proper. If I understood correct, contract is not found for given contract manager, not "not found" at all

File: HappyTravel.Hiroshima.Data/Extensions/DirectContractsDbContextExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,17 @@
+﻿using System.Collections.Generic;
+using Microsoft.EntityFrameworkCore;
+
+namespace HappyTravel.Hiroshima.Data.Extensions
+{
+    public static class DirectContractsDbContextExtensions
+    {
+        public static void DetachEntries<T>(this DirectContractsDbContext dbContext,  List<T> items)
```
Comment (hasmikharutyunyan) [inline HappyTravel.Hiroshima.Data/Extensions/DirectContractsDbContextExtensions.cs:8]: Maybe IEnumerable better than List?

File: HappyTravel.Hiroshima.DirectContracts/Services/Management/ContractManagementRepository.cs [hunk 5]
```diff
@@ -16,7 +16,7 @@ public ContractManagementRepository(DirectContractsDbContext dbContext)
         }
 
         
-        public async Task<Contract> GetContract(int contractManagerId, int contractId) =>
+        public async Task<Contract> GetContract(int contractId, int contractManagerId) =>
             await _dbContext.Contracts.SingleOrDefaultAsync(c => c.ContractManagerId == contractManagerId && c.Id == contractId);
```
Comment (hasmikharutyunyan) [inline HappyTravel.Hiroshima.DirectContracts/Services/Management/ContractManagementRepository.cs:20]: no need check for default ?

File: HappyTravel.Hiroshima.DirectManager/Services/ISeasonManagementService.cs [hunk 6]
```diff
@@ -1,5 +1,15 @@
...
+using CSharpFunctionalExtensions;
+
+namespace HappyTravel.Hiroshima.DirectManager.Services
 {
     public interface ISeasonManagementService
-    { }
+    {
+        Task<Result<List<Models.Responses.Season>>> Get(int contractId);
+        
+        Task<Result<List<Models.Responses.Season>>> Replace(int contractId, List<Models.Requests.Season> seasons);
```
Comment (kirillta) [inline HappyTravel.Hiroshima.DirectManager/Services/ISeasonManagementService.cs:11]: Don't quite understand what means _replace_ a season? As I understand it's _add_ seasons. Also, is replace mean we remove old seasons, why?

File: HappyTravel.Hiroshima.DirectManager/RequestValidators/SeasonValidator.cs [hunk 7]
```diff
@@ -0,0 +1,16 @@
+﻿using System;
+using FluentValidation;
+
+namespace HappyTravel.Hiroshima.DirectManager.RequestValidators
+{
+    public class SeasonValidator: AbstractValidator<Models.Requests.Season>
```
[the remark this answers is not in the document]
Comment (oleg1std) [reply HappyTravel.Hiroshima.DirectManager/RequestValidators/SeasonValidator.cs:6]: In my opinion, it's more convenient to have a separate validator for every request model. Or use validation attributes in request models.

File: HappyTravel.Hiroshima.DirectManager/Services/ISeasonManagementService.cs (hunk 6 above)
Comment (oleg1std) [reply HappyTravel.Hiroshima.DirectManager/Services/ISeasonManagementService.cs:11]: We have to upload all seasons at once. It's a more clear way to define seasons to the contracted period.
We have to remove old seasons to exclude intersections and other validation errors.

Comment (kirillta) [reply HappyTravel.Hiroshima.DirectManager/Services/ISeasonManagementService.cs:11]: That will not work

**Pull request merged**